.takeshape/pattern/schema.json

```diff
@@ -1,806 +1,1117 @@
+        "resolver": {
+          "results": {
+            "ops": [
+              {
+                "path": "$",
+                "mapping": [["get", {"path": "$resolvers.subscriptions"}]]
+              }
+            ]
+          },
+          "compose": [
             {
-              "path": "$",
-              "mapping": [
-                [
-                  "expressionEval",
+              "id": "customerByEmail",
+              "name": "rest:get",
+              "service": "recharge",
+              "path": "/customers",
+              "searchParams": {
+                "ops": [{"path": "email", "mapping": "$claims.email"}]
+              }
+            },
+            {
+              "id": "subscriptions",
+              "name": "rest:get",
+              "service": "recharge",
+              "path": "/subscriptions",
+              "searchParams": {
+                "ops": [
                   {
-                    "expression": "map(compose(set('prices', placeholder, newObject()), get(placeholder, groupBy('product', $resolvers.prices.data)), get('id')), $resolvers.products.data)"
+                    "path": "shopify_customer_id",
+                    "mapping": "$resolvers.customerByEmail.customers[0].shopify_customer_id"
                   }
-                ],
-                ["getMultiple", { "paths": ["$previousDirective", "$resolvers.products.data"] }],
-                ["zipAndMerge", {}]
-              ]
+                ]
+              }
             }
           ]
         },
-        "compose": [
-          {
-            "id": "products",
-            "name": "rest:get",
-            "service": "stripe",
-            "path": "/v1/products",
-            "searchParams": { "ops": [{ "path": "active", "value": "true" }] }
+        "description": "Get my subscriptions"
+      },
+      "getMyPayments": {
+        "shape": "ShopifyStorefront_Customer",
+        "resolver": {
+          "results": {
+            "ops": [
+              {
+                "path": "$",
+                "mapping": [["get", {"path": "$resolvers.getCustomerOrders"}]]
+              }
+            ]
           },
-          {
-            "id": "prices",
-            "name": "rest:get",
-            "service": "stripe",
-            "path": "/v1/prices",
-            "searchParams": { "ops": [{ "path": "active", "value": "true" }] }
-          }
-        ]
+          "compose": [
+            {
+              "id": "createShopifyCustomerAccessToken",
+              "name": "graphql:mutation",
+              "service": "shopify-storefront",
+              "fieldName": "customerAccessTokenCreate",
+              "options": {
+                "selectionSet": "{customerAccessToken{accessToken } customerUserErrors{field}}"
+              },
+              "args": {
+                "ops": [
+                  {"path": "input.email", "mapping": "$claims.email"},
+                  {"path": "input.password", "mapping": "$claims.azp"}
```

Comment:
You're giving every customer the same password here. Very bad. The admin API should be used for all customer interactions since we're authenticating them outside of Shopify and performing actions on their behalf.

For instance, in this case you're getting customer orders, so you'd read the email off their claims and use it as a filter on a this query: https://shopify.dev/api/admin-graphql/2022-01/queries/orders